### ISSUE TITLE
Confusion: handle vs. pin

### ISSUE THREAD

**Issue by nkueng** [printablealice/a3#8]
Hey Jim,

First of all thanks for the dope programming exercise. Once again really impressive stuff you guys let us play around with...

Now to my question: I am confused about the differences between what you refer to as "handles" and "pins". Based on the first part of the handout, I would think the two refer to the same thing. However, when switching from the fem- to the manip-app, I got confused since the method of interacting with the beam changed. 
So, in the context of the manip-app, what do you call the red dots that we can interact with? Pins? And what about the (dark?) red "elongated droplets" at the ends of the beam? Handles?

My question is mainly geared towards the regularizer exercise, where I would like to deal with mentioned "elongated droplets" but realized I don't know behind which variable name they hide.

Thanks a lot for your help!

**Comment by printablealice** [comment]
Oh for sure!--And good question :)
Please also see this answer [**here**](https://github.com/printablealice/a3/issues/5) which will go more in depth.
Here I will limit myself to a high level explanation.
tl;dr The handles are the red widgets you drag around, and together they serve as a parameterization of the anchor positions of the pins attached to the nodes on the ends of the bar.

---

First, let's forget about handles completely and talk pins.
A pin is a zero length spring that pins a a node to a point in world space.
Typically that spring has really high spring constant, so we could also say that really:
_A pin is a soft constraint on the position of a node._

Why do this?
Well, we want the node to be somewhere, but we don't want to bother with the hassle of either:
- Using a solver that can handle constraints (e.g. SQP)
- Removing the location of the that node from ```x```

These would both work, but so will adding a soft constraint, so we do that instead.

Okay great, so now we want to put the (left and right) ends of the bar somewhere, ideally right where they would be if they were being held by the grippers of a robot like in [**this paper**](https://www.youtube.com/watch?v=TrvnQ1Xe4c0).

At this point we have two options as I see it.

1. We could specify the location of each node on the ends of the bar individually (i.e. our control vector ```u``` would contain the anchor position of every single pin for the nodes on the ends of the bar.)  This is actually not a bad idea, you would need to add some regularizer that keeps you from stretching or squooshing the ends of the bar, but I bet it could be made to work without too much trouble.

Instead I went with

2. Specify a rigid translation and rotation for each end of the bar, and then use that to calculate the anchor position of the pins [**like this**](https://github.com/printablealice/a3/issues/5).  This keeps your control vector small (just the translation and rotation of the left end, and the translation and rotation of the right end).  I give you a little widget for each end of the bar, consisting of a red dot for specifying translation, and a red teardrop for specifying translation.  This widget is really what I'm calling a handle :)

Please note that you could also call the combination of all the widgets _the rig_ or _the parameterization of pin anchor positions_.

Let me know if there's anything in there that didn't make sense!

Best,
Jim

**Comment by nkueng** [comment]
Jim, I really appreciate the effort you put into every answer! Also, I am very sorry I started this thread before realizing that there actually exists an awesome intro video of you intuitively explaining what is going in the simulation. Thanks to all those resources, the difference between handles and pins finally came clear to me. Thanks a lot again :)

**Comment by printablealice** [comment]
Oh dope! :) Lol oh no worries it didn’t occur to me that maybe people
didn’t know about the video 😂 I’ll link it somewhere on the Github.
